repo/blob/azure/azure_storage_test.go

```diff
@@ -270,3 +274,134 @@ func TestAzureStorageInvalidCreds(t *testing.T) {
 		t.Errorf("unexpected success connecting to Azure blob storage, wanted error")
 	}
 }
+
+// TestAzureStorageImmutabilityProtection runs through the behavior of Azure immutability protection.
+func TestAzureStorageImmutabilityProtection(t *testing.T) {
+	t.Parallel()
+	testutil.ProviderTest(t)
+
+	// must be with ImmutableStorage with Versioning enabled
+	container := getEnvOrSkip(t, testContainerEnv)
+	storageAccount := getEnvOrSkip(t, testStorageAccountEnv)
+	storageKey := getEnvOrSkip(t, testStorageKeyEnv)
+
+	// create container if does not exist
+	createContainer(t, container, storageAccount, storageKey)
```

Comment:
note: this addition is basically pointless since we are not deleting/creating containers each time I assume; otherwise there is no point in having the cleanup step for data older than 1 hour since there wouldn't be any

Reply:
Yes, we should create the container once manually and then keep reusing it. Creating and deleting containers for this test may not be a good idea because we'll need to create unique containers and then we might end up in a situation where we start leaking these containers. Plus, deleting buckets can take longer if there are locked objects underneath.

Reply:
I think @julio-lopez said that he is preparing a script that could be run one-time during provisioning to setup the required buckets in Kopia test env which we can keep repurposing.

Reply:
Filed #3477 to track the container creation for these tests.